packages/ui/src/component/UICanvas.ts

```diff
@@ -0,0 +1,619 @@
+  @ignoreClone
+  _orderedRenderers: UIRenderer[] = [];
+  /** @internal */
+  @ignoreClone
+  _realRenderMode: number = CanvasRealRenderMode.None;
+  /** @internal */
+  @ignoreClone
+  _disorderedElements: DisorderedArray<IElement> = new DisorderedArray<IElement>();
+
+  @ignoreClone
+  private _renderMode = CanvasRenderMode.WorldSpace;
+  @ignoreClone
+  private _renderCamera: Camera;
+  @ignoreClone
+  private _cameraObserver: Camera;
+  @ignoreClone
+  private _resolutionAdaptationStrategy = ResolutionAdaptationStrategy.HeightAdaptation;
+  @ignoreClone
+  private _sortOrder: number = 0;
+  @ignoreClone
+  private _distance: number = 10;
+  @deepClone
+  private _referenceResolution: Vector2 = new Vector2(800, 600);
+  @deepClone
+  private _referenceResolutionPerUnit: number = 100;
+  @ignoreClone
+  private _hierarchyVersion: number = -1;
+
+  /**
+   * The conversion ratio between reference resolution and unit for UI elements in this canvas.
+   */
+  get referenceResolutionPerUnit(): number {
+    return this._referenceResolutionPerUnit;
+  }
+
+  set referenceResolutionPerUnit(value: number) {
+    if (this._referenceResolutionPerUnit !== value) {
+      this._referenceResolutionPerUnit = value;
+    }
+  }
+
+  /**
+   * The reference resolution of the UI canvas in `ScreenSpaceCamera` and `ScreenSpaceOverlay` mode.
+   */
+  get referenceResolution(): Vector2 {
+    return this._referenceResolution;
+  }
+
+  set referenceResolution(value: Vector2) {
+    const referenceResolution = this._referenceResolution;
+    if (referenceResolution === value) return;
+    (referenceResolution.x !== value.x || referenceResolution.y !== value.y) && referenceResolution.copyFrom(value);
+  }
+
+  /**
+   * The rendering mode of the UI canvas.
+   */
+  get renderMode(): CanvasRenderMode {
+    return this._renderMode;
+  }
+
+  set renderMode(mode: CanvasRenderMode) {
+    const preMode = this._renderMode;
+    if (preMode !== mode) {
+      this._renderMode = mode;
+      this._updateCameraObserver();
+      this._setRealRenderMode(this._getRealRenderMode());
+    }
+  }
+
+  /**
+   * The camera used to render the UI canvas in `ScreenSpaceCamera` mode.
+   * @remarks If set `ScreenSpaceCamera` but no corresponding camera is assigned, the actual rendering mode defaults to `ScreenSpaceOverlay`.
+   */
+  get renderCamera(): Camera {
+    return this._renderCamera;
+  }
+
+  set renderCamera(value: Camera) {
+    const preCamera = this._renderCamera;
+    if (preCamera !== value) {
+      this._renderCamera = value;
+      this._updateCameraObserver();
+      this._setRealRenderMode(this._getRealRenderMode());
+    }
+  }
+
+  /**
+   * The screen resolution adaptation strategy of the UI canvas in `ScreenSpaceCamera` and `ScreenSpaceOverlay` mode.
+   */
+  get resolutionAdaptationStrategy(): ResolutionAdaptationStrategy {
+    return this._resolutionAdaptationStrategy;
+  }
+
+  set resolutionAdaptationStrategy(value: ResolutionAdaptationStrategy) {
```

Comment:
Why not call `resolutionAdaptationMode`?

Reply:
Done.